drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -950,16 +964,22 @@ int bnxt_hwrm_vnic_qcfg(struct bnxt *bp, struct bnxt_vnic_info *vnic,
 	HWRM_CHECK_RESULT;
 
 	vnic->dflt_ring_grp = rte_le_to_cpu_16(resp->dflt_ring_grp);
-	vnic->fw_rss_cos_lb_ctx = rte_le_to_cpu_16(resp->rss_rule);
-	// vnic->cos_rule = rte_le_to_cpu_16(resp->cos_rule);
-	// vnic->lb_rule = rte_le_to_cpu_16(resp->lb_rule);
+	vnic->rss_rule = rte_le_to_cpu_16(resp->rss_rule);
+	vnic->cos_rule = rte_le_to_cpu_16(resp->cos_rule);
+	vnic->lb_rule = rte_le_to_cpu_16(resp->lb_rule);
 	vnic->mru = rte_le_to_cpu_16(resp->mru);
 	vnic->func_default = rte_le_to_cpu_32(
 			resp->flags) & HWRM_VNIC_QCFG_OUTPUT_FLAGS_DEFAULT;
 	vnic->vlan_strip = rte_le_to_cpu_32(
 			resp->flags) & HWRM_VNIC_QCFG_OUTPUT_FLAGS_VLAN_STRIP_MODE;
 	vnic->bd_stall = rte_le_to_cpu_32(
 			resp->flags) & HWRM_VNIC_QCFG_OUTPUT_FLAGS_BD_STALL_MODE;
+	vnic->roce_dual = rte_le_to_cpu_32(
+			resp->flags) & HWRM_VNIC_QCFG_OUTPUT_FLAGS_ROCE_DUAL_VNIC_MODE;
+	vnic->roce_only = rte_le_to_cpu_32(
+			resp->flags) & HWRM_VNIC_QCFG_OUTPUT_FLAGS_ROCE_ONLY_VNIC_MODE;
```

Comment:
Because the flags are always applied, we need to track all of them, not just the ones we use.  We may want to go to a bitmask in the future, but the QCFG and CFG bits are not guaranteed to be in sync.